tcms/core/widgets.py

```diff
@@ -40,3 +40,35 @@ class Media:
             "prismjs/plugins/autoloader/prism-autoloader.min.js",
             "js/simplemde_security_override.js",
         ]
+
+
+class DurationWidget(forms.Textarea):
+
+    def render(self, name, value, attrs=None, renderer=None):
+        rendered_duration = super().render(name, value, attrs, renderer)
+        rendered_duration += """
+<input id="%s" type="text" style="display: none">
+<script>
+$(function () {
+    $('#%s').durationPicker({
+      showDays: true,
+      showHours: true,
+      showMinutes: true,
+      showSeconds: true,
+    });
+});
+
+</script>
+""" % (
+            attrs['id'],
+            attrs['id'],
+        )
```

Comment:
FTR the HTML snippet here does exactly the same thing as duratyion_picker_script.html which is currently unused.  See https://docs.djangoproject.com/en/3.2/intro/tutorial03/#a-shortcut-render for how to render a template file to string and passing context variables to the template file.